src/collective/volto/formsupport/utils.py

```diff
@@ -37,3 +39,12 @@ def get_blocks(context):
     flat = list(flatten_block_hierachy(blocks))
 
     return dict(flat)
+
+
+def generate_email_token(uid, email):
+    """Generates the email verification token"""
+    from zope.component import getUtility
+
+    manager = getUtility(IKeyManager)
+
+    return md5((uid + email + manager.secret()).encode()).hexdigest()[:10]
```

Comment:
md5 non è OTP anche se il secret ruota periodicamente (non so dirti con che tempi/logica), devi inserire una forma di scadenza puoi usare direttamente https://github.com/pyauth/pyotp?tab=readme-ov-file#time-based-otps usando come secret i dati che usavi qui. Per il resto tutto ok per me.

Reply:
Si, infatti stavo pensado che il secret cambia, in caso si può anche usare il datetime slottato per 15 min for example

Reply:
done